File: java/DataFrame/src/main/java/coyote/dataframe/S32Type.java
package coyote.dataframe;

import coyote.util.ByteUtil;

/** (8) Type code representing a signed, 32-bit value in the range of -2,147,483,648 to 2,147,483,647 */
public class S32Type implements FieldType
{
  private static final int _size = 4;

  private final static String _name = "S32";




  public boolean checkType( Object obj )
  {
    return obj instanceof Integer;
  }




  public Object decode( byte[] value )
  {
    return new Integer( ByteUtil.retrieveInt( value, 0 ) );
  }




  public byte[] encode( Object obj )
  {
    return ByteUtil.renderInt( ( (Integer)obj ).intValue() );
  }




  public String getTypeName()
  {
    return _name;
  }




  public boolean isNumeric()
  {
    return true;
  }




  public int getSize()
  {
    return _size;
  }

}

File: csharp/DataFrame/Coyote/DataFrame/S32Type.cs
#region license
// Copyright (c) 2006 Stephan D. Cote' - All rights reserved.
// 
// This program and the accompanying materials are made available under the 
// terms of the MIT License which accompanies this distribution, and is 
// available at http://creativecommons.org/licenses/MIT/
// *
// Contributors:
//   Stephan D. Cote 
//      - Initial API and implementation
#endregion
using System;

namespace Coyote.DataFrame {



    /// This is an Integer, representing a signed, 32-bit value in the range of -2,147,483,648 to 2,147,483,647 
    public class S32Type : FieldType {
        private const int _size = 4;

        private const string _name = "S32";



        public virtual string TypeName {
            get {
                return _name;
            }
        }




        public virtual bool IsNumeric {
            get {
                return true;
            }
        }




        public virtual int Size {
            get {
                return _size;
            }
        }


        public virtual bool CheckType( object obj ) {
            return obj is int;
        }




        public virtual object Decode( byte[] @value ) {
            return ByteUtil.RetrieveInt( @value, 0 );
        }




        public virtual byte[] Encode( object obj ) {
            return ByteUtil.RenderInt( (int)obj );
        }





        public virtual string StringValue( byte[] val ) {
            if ( val == null ) {
                return "";
            } else {
                object obj = Decode( val );
                if ( obj != null )
                    return obj.ToString();
                else
                    return "";
            }
        }

    }

}